java/util/Base64/Base64Test.cpp

```diff
@@ -0,0 +1,313 @@
+
+    ASSERT_TRUE(Arrays::equals(expectedResultFromInput4BytesWith0BytePadding,
+                               resultFromInput4BytesWith0BytePadding));
+    ASSERT_TRUE(Arrays::equals(expectedResultFromInput4BytesWith1BytePadding,
+                               resultFromInput4BytesWith1BytePadding));
+    ASSERT_TRUE(Arrays::equals(resultFromInput4BytesWith2BytesPadding,
+                               resultFromInput4BytesWith2BytesPadding));
+
+    // Basic decoder without padding character '=' in input.
+    Array<byte> inputWithoutPadding = {'T', 'W', 'E'};  // ... with padding {'T', 'W', 'E', '='}
+    Array<byte> resultFromInputWithoutPadding =
+            basicDecoder.decode(inputWithoutPadding);
+    Array<byte> expectedResultFromInputWithoutPadding = {'M', 'a'};
+    ASSERT_TRUE(Arrays::equals(expectedResultFromInputWithoutPadding,
+                               resultFromInputWithoutPadding));
+
+    // Passing output Array<byte> by reference.
+    Array<byte> inputArray =
+            {'V', '2', 'V', 's', 'Y', '2', '9', 't', 'Z', 'S',
+             'B', '0', 'b', 'y', 'B', 'W', 'a', 'W', 'V', '0',
+             'b', 'm', 'F', 't', 'I', 'Q', '=', '='};
+    Array<byte> outputArrayNotEnoughSize(10);
+    try {
+        basicDecoder.decode(inputArray, outputArrayNotEnoughSize);
+    } catch (IllegalArgumentException ex) {
+        ASSERT_STR("Output byte array is too small for decoding all input bytes",
+                   ex.getMessage().toString());
+    }
+    Array<byte> outputArray(100);
+    int index;
+    for (index = 0; index < outputArray.length; index++) { // Filling with '*'.
+        outputArray[index] = '*';
+    }
+    int realLength = basicDecoder.decode(inputArray, outputArray);
+    String expectedOutputAsString = "Welcome to Vietnam!";
+    Array<byte> expectedOutputArray =
+            {'W', 'e', 'l', 'c', 'o', 'm', 'e', ' ', 't', 'o',
+             ' ', 'V', 'i', 'e', 't', 'n', 'a', 'm', '!', '*',
+             '*', '*', '*', '*', '*', '*', '*', '*', '*', '*',
+             '*', '*', '*', '*', '*', '*', '*', '*', '*', '*',
+             '*', '*', '*', '*', '*', '*', '*', '*', '*', '*',
+             '*', '*', '*', '*', '*', '*', '*', '*', '*', '*',
+             '*', '*', '*', '*', '*', '*', '*', '*', '*', '*',
+             '*', '*', '*', '*', '*', '*', '*', '*', '*', '*',
+             '*', '*', '*', '*', '*', '*', '*', '*', '*', '*',
+             '*', '*', '*', '*', '*', '*', '*', '*', '*', '*'}; // "Welcome to Vietnam!"
+    ASSERT_EQUAL(expectedOutputAsString.length(), realLength);
+    ASSERT_TRUE(Arrays::equals(expectedOutputArray, outputArray));
+
+    // Input as String.
+    String inputAsString = "V2VsY29tZSB0byBWaWV0bmFtIQ==";
+    Array<byte> resultOfInputAsString = basicDecoder.decode(inputAsString);
+    Array<byte> expectedResultOfInputAsString =
+            {'W', 'e', 'l', 'c', 'o', 'm', 'e', ' ', 't', 'o',
+             ' ', 'V', 'i', 'e', 't', 'n', 'a', 'm', '!'}; // "Welcome to Vietnam!"
```

Comment:
At least two space between code and comment is best. You should try :)
You can shorten or split this line. I think it will be better for looking :D